framework/src/main/java/org/apache/drill/test/framework/Utils.java

```diff
@@ -169,7 +170,21 @@ public static ConnectionPool getConnectionPool() {
         boolean foundTests = false;
         for (String testGroup : testGroups) {
           if (categories != null && !categories.contains(testGroup)) {
-            continue;
+            if (categories.contains("smoke") &&
+                testGroup.equals("smoke_simba_jdbc") &&
+                !testDefFile.getName().endsWith("apache.json") &&
+                !testDefFile.getName().endsWith("simba.jdbc.json")) {
+              // this json file applies to all Drivers.
+              // this json file is used for smoke tests
+              // this json file has not been modified to include
+              //    smoke_simba_jdbc tests
+              // therefore assume this json file should be modified to include
+              //    smoke_simba_jdbc tests
+              // therefore check for smoke_simba_jdbc tests
```

Comment:
I thought it would be more helpful if the comments could explain the code step-by-step.  But I can write a general comment.

I was wondering where to document the test definition file types.  Perhaps this is a good place for it.

Yes, I can rename apache.json to apache.jdbc.json.

Backward compatibilty is maintained.  By default, our current json files stay.  I have added two more categories (for jdbc and odbc) to each json file.  Most tests pass without any changes for Apache and Simba drivers.  A small number do not.  For these tests, I have renamed the query files to use a new file extension (*.q2) and created new json files for them.  There is a json file that Apache uses for these special tests, and a json file that Simba uses for these special tests.  Apache expected results files remain as they were.  I added new Simba expected results files with a new file extension (*.sjdbc.e_tsv as an example).

Having said that, I am concerned it is error-prone to require new json files to have additional categories for Simba.  So I added this code to enable existing json files to run with Simba without any changes.  When we had our last discussions, this was considered not the best approach.  So I have implemented both, so we can see what the impact is.

Perhaps I should add a few json files and test files so you can see how the whole thing works together.  Initially I just put the code here because it seems useful to have code separated from the rest of the changes, when the changes are large.